pkg/scheduler/reconciliation/inmemory.go

```diff
@@ -106,6 +106,18 @@ func (r *InMemoryReconciliationRepository) RemoveReconciliation(schedulingID str
 	return nil
 }
 
+func (r *InMemoryReconciliationRepository) RemoveReconciliations(reconEntities []*model.ReconciliationEntity) error {
+	r.mu.Lock()
+	defer r.mu.Unlock()
+
+	for _, recon := range reconEntities {
+		delete(r.reconciliations, recon.RuntimeID)
```

Comment:
I think the entire implementation could be just iteration over schedulingIDs and delegating the work to the already existing method.
This may look inefficient, but the in-memory implementation is very fast and it's used only in tests AFAIK.
Considering the above, I'd change the signature of this method to only require a slice of `schedulingIDs`.

Reply:
done